### PR TITLE
Aducm i2c_ops added

### PR DESCRIPTION
**Pull request by APorumb**
-added i2c platform ops for Aducm platform.

-fixed projects build errors

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
One question on my side: were you able to test the projects with the new i2c ops implementation?

**Comment by APorumb** [comment]
> 
> 
> One question on my side: were you able to test the projects with the new i2c ops implementation?

I created a new project to test it and it worked as expected. I did not test it on older projects.

**Review by amiclaus — APPROVED** [review]
lgtm, as long as we test on at least one project that is already on the no-OS repo.

File: drivers/platform/aducm3029/aducm3029_i2c.c [hunk 1]
```diff
@@ -74,17 +75,6 @@ static uint32_t last_bitrate = UNINITIALIZED_BITRATE;
...
- * @struct i2c_extra
- * @brief I2C platform specific parameters.
- */
-struct aducm_i2c_extra {
-	/** Pointer to the prologue data. */
-	uint8_t *prologue_data;
-	/** Prologue size. */
-	uint8_t prologue_size;
-};
-
```
Comment (mchindri) [inline drivers/platform/aducm3029/aducm3029_i2c.c:87]: There is no reason to move this to the .h It is only used here. No external code need to access this fields

File: drivers/platform/aducm3029/aducm3029_i2c.c [hunk 2]
```diff
@@ -126,8 +116,8 @@ static uint32_t set_transmission_configuration(struct i2c_desc *desc)
  * it is not used and must be set to NULL.
  * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
  */
-int32_t i2c_init(struct i2c_desc **desc,
-		 const struct i2c_init_param *param)
+int32_t aducm3029_i2c_init(struct i2c_desc **desc,
```
Comment (mchindri) [inline drivers/platform/aducm3029/aducm3029_i2c.c:119]: make functions static. The should only be used through ops.

Comment (mchindri) [reply drivers/platform/aducm3029/aducm3029_i2c.c:119]: nitpick. If function became static, rename is not needed, but this is not a big issue anyway

File: drivers/platform/aducm3029/i2c_extra.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+*******************************************************************************/
+
+#ifndef I2C_EXTRA_H_
+#define I2C_EXTRA_H_
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
```
Comment (mchindri) [inline drivers/platform/aducm3029/i2c_extra.h:47]: If removing aducm_i2c_extra  this won't be needed

File: drivers/platform/aducm3029/aducm3029_i2c.c (hunk 1 above)
Comment (APorumb) [reply drivers/platform/aducm3029/aducm3029_i2c.c:87]: The idea was to keep the code consistent (like any other _extra.h file).

File: drivers/platform/aducm3029/aducm3029_i2c.c (hunk 2 above)
Comment (APorumb) [reply drivers/platform/aducm3029/aducm3029_i2c.c:119]: Same, wanted to keep the code consistent. Function definitions are not included in the header file so they should be used only through ops.

Comment (mchindri) [reply drivers/platform/aducm3029/aducm3029_i2c.c:119]: This means that they should be static :)

**Comment by APorumb** [comment]
V2:

-reworked commit history to not have failed tests for each of the commits
-made aducm_i2c functions static

**Review by mchindri — APPROVED** [review]

**Pull request merged**